webserver/views/api_compat.py

```diff
@@ -0,0 +1,352 @@
+            text(token.token)
+    return format_response('<?xml version="1.0" encoding="utf-8"?>\n' + yattag.indent(doc.getvalue()),
+                           output_format)
+
+
+def get_session(request, data):
+    """ Create new session after validating the API_key and token.
+    """
+    output_format = data.get("format", "xml")
+    try:
+        api_key = data['api_key']
+        token = Token.load(data['token'], api_key)
+        if token.has_expired():
+            raise InvalidAPIUsage(15, output_format=output_format)  # Token expired
+    except KeyError:
+        raise InvalidAPIUsage(6, output_format=output_format)       # Missing Required Params
+
+    if not token:
+        if not Token.is_valid_api_key(api_key):
+            raise InvalidAPIUsage(10, output_format=output_format)  # Invalid API_key
+        raise InvalidAPIUsage(4, output_format=output_format)       # Invalid token
+
+    if not token.user:
+        raise InvalidAPIUsage(14, output_format=output_format)      # Unauthorized token
+
+    session = Session.create(token)
+
+    doc, tag, text = Doc().tagtext()
+    with tag('lfm', status="ok"):
+        with tag('session'):
+            with tag('name'):
+                text(session.user.name)
+            with tag('key'):
+                text(session.sid)
+            with tag('subscriber'):
+                text('0')
+
+    return format_response('<?xml version="1.0" encoding="utf-8"?>\n' + yattag.indent(doc.getvalue()),
+                           data.get('format', "xml"))
+
+
+def _to_native_api(lookup, method="track.scrobble"):
+    """ Converts the list of listens received in the new Last.fm submission format
+        to the native ListenBrainz API format.
+        Returns: type_of_listen and listen_payload
+    """
+    listen_type = "listens"
+    if method == "track.updateNowPlaying":
+        listen_type = "playing_now"
+        if len(lookup.keys()) != 1:
+            raise InvalidAPIUsage(6, output_format=output_format)       # Invalid parameters
+
+    listens = []
+    for ind, data in lookup.iteritems():
+        listen = {
+            'track_metadata': {
+                'additional_info': {}
+            }
+        }
+        if 'timestamp' in data:
+            listen['listened_at'] = data['timestamp']
+        if 'album' in data:
+            listen['track_metadata']['release_name'] = data['album']
+        if 'track' in data:
+            listen['track_metadata']['track_name'] = data['track']
+        if 'artist' in data:
+            listen['track_metadata']['artist_name'] = data['artist']
+        if 'mbid' in data:
+            listen['track_metadata']['release_mbid'] = data['mbid']
+        if 'trackNumber' in data or 'duration' in data or 'albumArtist' in data or \
+                'choosenByUser' in data or 'streamId' in data or 'context' in data:
+            # Album artist, Song Duration, trackNumber on the album, etc
+            # are not supported by the native ListenBrainz API
```

Comment:
A lot a params are mentioned on the website (http://www.last.fm/api/show/track.scrobble).
I have added support for all of them in LB, except for album_artist as we do not accept album artist name in the data.